extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsHandler.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.smallrye.metrics.runtime;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import javax.enterprise.inject.spi.CDI;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+
+import io.smallrye.metrics.MetricsRequestHandler;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+
+public class SmallRyeMetricsHandler implements Handler<RoutingContext> {
+
+    private final String metricsPath;
+
+    private static final Logger LOGGER = Logger.getLogger(SmallRyeMetricsHandler.class.getName());
+
+    public SmallRyeMetricsHandler() {
+        metricsPath = ConfigProvider.getConfig()
+                .getOptionalValue("quarkus.smallrye-metrics.path", String.class)
+                .orElse("/metrics");
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        MetricsRequestHandler internalHandler = CDI.current().select(MetricsRequestHandler.class).get();
+        HttpServerResponse response = routingContext.response();
+        HttpServerRequest request = routingContext.request();
+        Stream<String> acceptHeaders = request.headers().getAll("Accept").stream();
+
+        try {
+            internalHandler.handleRequest(request.path(), metricsPath, request.rawMethod(), acceptHeaders,
```

Comment:
Is the logic inside the `internalHandler.handleRequest()` non-blocking? If so then OK, otherwise you should use `HandlerType#BLOCKING` instead.

Reply:
Ok I added `HandlerType#BLOCKING`

Reply:
I think it's safer to use "blocking" for now. While it could be non-blocking, it the user declare a metric where the computation is blocking that could have a terrible impact.